routing/http/server/filters.go

```diff
@@ -0,0 +1,201 @@
+package server
+
+import (
+	"errors"
+	"reflect"
+	"slices"
+	"strings"
+
+	"github.com/ipfs/boxo/routing/http/types"
+	"github.com/ipfs/boxo/routing/http/types/iter"
+	"github.com/multiformats/go-multiaddr"
+)
+
+// filters implements IPIP-0484
+
+func parseFilter(param string) []string {
+	if param == "" {
+		return nil
+	}
+	return strings.Split(strings.ToLower(param), ",")
+}
+
+// applyFiltersToIter applies the filters to the given iterator and returns a new iterator.
+func applyFiltersToIter(recordsIter iter.ResultIter[types.Record], filterAddrs, filterProtocols []string) iter.ResultIter[types.Record] {
+	mappedIter := iter.Map(recordsIter, func(v iter.Result[types.Record]) iter.Result[types.Record] {
+		if v.Err != nil || v.Val == nil {
```

Comment:
Why is it OK to ignore the error if `v.Val != nil`? Please comment.

Reply:
If `v.Val != nil && v.Val != nil` , the result value will be untouched. The idea here was to avoid any processing if there's an error or a nil value. 

Does that make sense?

Reply:
yes